timeseries/src/autogluon/timeseries/models/ensemble/greedy_ensemble.py

```diff
@@ -1,226 +1,107 @@
 import copy
 import logging
-import time
-from typing import List
+from typing import Dict, List
 
 import numpy as np
 
+import autogluon.core as ag
 from autogluon.core.models.greedy_ensemble.ensemble_selection import AbstractWeightedEnsemble, EnsembleSelection
 from autogluon.timeseries import TimeSeriesDataFrame
+from autogluon.timeseries.evaluator import TimeSeriesEvaluator
 from autogluon.timeseries.models.abstract import AbstractTimeSeriesModel
 
 logger = logging.getLogger(__name__)
 
 
 class TimeSeriesEnsembleSelection(EnsembleSelection):
-    def __init__(  # noqa
+    def __init__(
         self,
         ensemble_size: int,
-        problem_type: str,
-        metric,
-        higher_is_better=False,
+        metric: TimeSeriesEvaluator,
+        problem_type: str = ag.constants.QUANTILE,
         sorted_initialization: bool = False,
         bagging: bool = False,
         tie_breaker: str = "random",
         random_state: np.random.RandomState = None,
         **kwargs,
     ):
-        # TODO: call super init?
-        self.ensemble_size = ensemble_size
-        self.problem_type = problem_type
-        self.metric = metric
-        self.higher_is_better = higher_is_better
-        self.sorted_initialization = sorted_initialization
-        self.bagging = bagging
-        self.use_best = True
-        if tie_breaker not in ["random", "second_metric"]:
-            raise ValueError(f"Unknown tie_breaker value: {tie_breaker}. Must be one of: ['random', 'second_metric']")
-        self.tie_breaker = tie_breaker
-        if random_state is not None:
-            self.random_state = random_state
-        else:
-            self.random_state = np.random.RandomState(seed=0)
-        self.quantile_levels = kwargs.get("quantile_levels", None)
-
-    def fit(self, predictions, labels, time_limit=None, identifiers=None, sample_weight=None):
-        self.ensemble_size = int(self.ensemble_size)
-        if self.ensemble_size < 1:
-            raise ValueError("Ensemble size cannot be less than one!")
-        self._fit(
-            predictions=predictions,
+        super().__init__(
+            ensemble_size=ensemble_size,
+            metric=metric,
+            problem_type=problem_type,
+            sorted_initialization=sorted_initialization,
+            bagging=bagging,
+            tie_breaker=tie_breaker,
+            random_state=random_state,
+            **kwargs,
+        )
+
+    def _fit(self, predictions, labels, time_limit=None, sample_weight=None):
+        self.dummy_pred = copy.deepcopy(predictions[0])
```

Comment:
This is because `dummy_pred` is introduced only in this class, and the superclass (`EnsembleSelection`) doesn't know to provide `dummy_pred` to this class' `_calculate_regret`. Regarding the memory issue, I'm adding 
```
self.dummy_pred = None
```
to make sure after fitting the instance variable is cleared 👍